Java/core/src/main/java/com/amazon/randomcutforest/tree/NodeStoreLarge.java

```diff
@@ -151,18 +146,20 @@ public void deleteInternalNode(int index) {
         if (parentIndex != null) {
             parentIndex[index] = capacity;
         }
-        if (pointSum != null) {
-            invalidatePointSum(index);
-        }
-        int idx = translate(index);
-        if (idx != Integer.MAX_VALUE) {
-            rangeSumData[idx] = 0.0;
-        }
         freeNodeManager.releaseIndex(index);
     }
 
     public int getMass(int index) {
-        return (isLeaf(index)) ? getLeafMass(index) : mass[index] != 0 ? mass[index] : (capacity + 1);
+        return mass[index] != 0 ? mass[index] : (capacity + 1);
```

Comment:
Mass is the number of leaves which are descendants of a node. For any reachable (from root) node, mass cannot be 0; so the value 0 is used to represent (capacity+1).

Reply:
Possibly in a future refactoring, I think it would be helpful to group together all methods that perform mass calculations, possibly into a utility class, and make the mass value only accessible through methods.